lib/Onyx.d.ts

```diff
@@ -0,0 +1,243 @@
+import {Keys, Values} from '.';
+import * as Logger from './Logger';
+
+type KeyList = keyof Keys;
+
+declare type ConnectOptions<K extends KeyList> = {
+    key: K;
+    statePropertyName?: string;
+    withOnyxInstance?: any; // TODO: Type this.
+    callback?: (value: Values[K] | undefined) => void;
+    initWithStoredValues?: boolean;
+    waitForCollectionCallback?: boolean;
+    selector?: any; // TODO: Type this.
```

Comment:
```suggestion
    selector?: (value: Values[K] | undefined) => Partial<Values[K]>
```

Reply:
@blazejkustra It must be possible to use `selector` with a `string` as well.

```
 * @param {String|Function} [mapping.selector] THIS PARAM IS ONLY USED WITH withOnyx(). If included, this will be used to subscribe to a subset of an Onyx key's data.
 *       If the selector is a string, the selector is passed to lodashGet on the sourceData. If the selector is a function, the sourceData and withOnyx state are
 *       passed to the selector and should return the simplified data. Using this setting on `withOnyx` can have very positive performance benefits because the component
 *       will only re-render when the subset of data changes. Otherwise, any change of data on any property would normally cause the component to re-render (and that can
 *       be expensive from a performance standpoint).
```

Reply:
This should do:
```ts
selector?: (value: Values[K]) => Partial<Values[K]> | string
or
selector?: (value: Values[K] | undefined) => Partial<Values[K] | undefined> | string
```

Reply:
> Also, I think it has to be => Values[K] or => Values[K] | undefined if I need to return the same structure the key has 🤔

@fabioh8010 The `selector` is used to get only get certain properties from a object that is mapped to the key passed.

You can watch 11:55 of [the presentation video](https://drive.google.com/file/d/1jvVk57tIwu0JFGScVBuMfaF1_J_x1zXX/view) for the example how this selector is used. In the video, selector is passed to `withOnyx` but `withOnyx`, I believe` uses `Onyx.connect` under the hood.

`selector`'s parameter type can be typed stricter. I created [a TS playground example for it](https://www.typescriptlang.org/play?ssl=17&ssc=35&pln=1&pc=1#code/JYWwDg9gTgLgBAbzgBQIa2KgNgEQKZ5hwC+cAZlBCHAOQwCeYeAtGXgM4w0DcAUA0zj5CAaTz0A8mQA8AFQB8cALxxZcPAA8YeAHYATdnAgAjAFZ4AxvAD8iOAG0RcYDrgBrcRDKq4AMjgAFJxQLgDmcAA+cDoAriDGeFAAlAC6AFyqjinqWroGcABKltB60sFhADRwMTpuOhAA7jqKtgAGACQIIsQAdJ3CYGKSMrJZ8sStkXAiGSLcJPYe9F4+-kEwITrhUbHxialwGbJ8-Ix40+LKtEsAjDRTNEsATPcnAucAatgxHFcIvHB3OIbhljBAIFg8KgdHxAc8MkgyOCMuUtlVjOgEXAMQAvFEbMIkKoARxiGgyuwSUHs2WIfGIvFOggkYBgwAgOnYciGOW0+kMQ0UKn+cPERyGFQBcAsxgyXywP3Y9lkQ2ymj5+RM5iscFsAQCADdvng5calSrxCkkspFGgMNgBtJ5Yrlar5NaogMhlInWbXZbFBS8AbEvTGe8LvQbiyYBzDCoY+zOdJHsCaPI+BGhk8Y3Grom4ynnunuEA).

I created `DeepKeyOf` utility method that unions of all possible paths

Reply:
taking a look at this..

Reply:
Thanks @hayata-suenaga for the suggestions, it worked very well and added some type-safety to string selectors 🎉